app/utils/validators/forms/FieldValidator.scala

```diff
@@ -44,6 +46,12 @@ object FieldValidator {
     def or(second: Boolean): Boolean = first || second
   }
 
+  val validEori: String => Boolean = (input: String) => input.matches("^GB[0-9]{1,15}$")
```

Comment:
We could use tiny types like case class Eori(val: String) extends AnyVal.  Then put the valid Regex with the class.  Then we don't have the possibility of another String ending up there and we have the validation with the thing being validated.  An easy fix that would save a lot of bugs ?

Reply:
I am a fan of dynamic typed functional languages :)
This function solves the problem at hand.